TypeSystem/flow-analysis/reachability_try_finally_A04_t03.dart

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion try finally: If `N` is a try/finally statement of the form
+/// `try B1 finally B2` then:
+/// - Let `before(B1) = split(before(N))`
+/// - Let `before(B2) = split(join(drop(after(B1)),
+///     conservativeJoin(before(N), assignedIn(B1), capturedIn(B1))))`
+/// - Let `after(N) = restrict(after(B1), after(B2), assignedIn(B2))`
+///
+/// @description Checks that if a type `T` is made a type of interest in a catch
+/// part of `B1` then it can be promoted in `B1`, `B2` and `after(N)`.
```

Comment:
I don't think `B1` can include a catch part. I think we have rules about `try .. catch .. catch` and about `try .. finally`, but the rule about `try .. catch .. catch .. finally` is missing.

So we're testing `try`-`catch`-`finally` statements as a separate kind of case, noting that these tests must be revisited when the spec rule is in place.